RecoEgamma/EgammaElectronProducers/plugins/GsfElectronProducer.cc

```diff
@@ -56,22 +56,22 @@ namespace {
       const auto& dnn_ele_pfid = hoc->iElectronDNNEstimator->evaluate(electrons, tfSessions);
       int jele = 0;
       for (auto& el : electrons) {
-        const auto& values = dnn_ele_pfid[jele];
+        const auto& [iModel, values] = dnn_ele_pfid[jele];
         // get the previous values
         auto& mvaOutput = mva_outputs[jele];
 
-        if (abs(el.superCluster()->eta()) <= extetaboundary) {
-          mvaOutput.dnn_e_sigIsolated = values[0];
-          mvaOutput.dnn_e_sigNonIsolated = values[1];
-          mvaOutput.dnn_e_bkgNonIsolated = values[2];
-          mvaOutput.dnn_e_bkgTau = values[3];
-          mvaOutput.dnn_e_bkgPhoton = values[4];
-        } else {
-          mvaOutput.dnn_e_sigIsolated = values[0];
+        if (iModel <= 3) {  // models 0,1,2,3 have 5 outpus in this version
+          mvaOutput.dnn_e_sigIsolated = values.at(0);
```

Comment:
each `at` call will check the size of the container. That isn't the most efficient. Instead I'd suggest adding
`assert(values.size() == 5)` at the beginning of the if and then just use `[]`.

Reply:
I implemented the assert and removed the `.at()`. We wanted a way to be sure that the code crashes if there is a model index misconfiguration, and the assert is a good choice. Thanks.